De-duplication of DayOffset and TextSize implementations

ebraminio opened the pull request:
It adds an animation to the value slider which I can take out if isn't desired as I think de-duplication worths the hassle regardless,

https://github.com/user-attachments/assets/cd12f039-fc8d-428d-8038-1d91976da0ac

ebraminio committed:
De-duplicate DayOffset and TextSize implementations

ebraminio committed:
Use locale aware number format

So when a locale that uses numerals other than western Arabic are
supported this would support it ootb.

ebraminio committed:
Add an animation for the value slider

amrbashir commented on app/src/main/res/values-fa/strings.xml:
Let's keep this translation file, in a separate PR. Also make sure to add an entry in CHANGELOG.md for it

ebraminio replied:
Oh this translation was completely wrong, it was just to test numeral formatting for locale which needed it, removed it

amrbashir commented on app/src/main/kotlin/me/amrbashir/hijriwidget/preference_activity/composables/DayOffset.kt:
I feel like this file could be deleted now, it can be embedded directly in `PreferencesListScreen.kt`, don't you think?

ebraminio replied:
Done

amrbashir commented on app/src/main/kotlin/me/amrbashir/hijriwidget/preference_activity/composables/TextSize.kt:
Same here

ebraminio replied:
Done

amrbashir reviewed (commented):
Thank you, the animations look great

ebraminio committed:
Inline DayOffset and TextSize

Their implementation has become that small that it doesn't make sense
to keep them on their own file and function anymore.

amrbashir reviewed (approved):
Thank you
